### PR TITLE
Update visualboyadvance-m.json

### LINKED ISSUE #347: visualboyadvance-m@2.1.4: make portable
You can make visualboyadvance-m portable by placing vbam.ini  in the installation folder of it.
C: \Users\user\AppData\Local\visualboyadvance-m\vbam.ini is no longer required.

### PR DESCRIPTION
**Pull request by satorope**
Make portable

Closes #347

### CODE REVIEW

**Comment by UnicorNora** [comment]
Can we merge this?

**Comment by beyondmeat** [comment]
@UnicorNora  Not in it's current state. Needs to be refactored to use pre_install as it's much cleaner than the method used here.

File: bucket/visualboyadvance-m.json [hunk 1]
```diff
@@ -13,16 +13,33 @@
             "hash": "e73cb3d4a0f6eb890549f04fef575ff4d9d3beb3b20b4877c4039cc2d9342311"
         }
     },
+    "post_install": [
+        "if (!(Test-Path \"$persist_dir\\vbam.ini.bak\")) {",
+        "   Set-Content \"$dir\\vbam.ini\" '[General]'",
+        "   Add-Content \"$dir\\vbam.ini\" 'OnlineUpdates=never'",
+        "   'Battery', 'Recording', 'Screenshot', 'State' | ForEach-Object {",
+        "   Add-Content -Path \"$dir\\vbam.ini\" -Value $_`Dir`=$_ }",
+        "} else {",
```
Comment (beyondmeat) [inline bucket/visualboyadvance-m.json:22]: else block can be dropped

File: bucket/visualboyadvance-m.json [hunk 2]
```diff
@@ -13,16 +13,33 @@
             "hash": "e73cb3d4a0f6eb890549f04fef575ff4d9d3beb3b20b4877c4039cc2d9342311"
         }
     },
+    "post_install": [
```
Comment (beyondmeat) [inline bucket/visualboyadvance-m.json:16]: make this a pre_install script

File: bucket/visualboyadvance-m.json [hunk 3]
```diff
@@ -13,16 +13,33 @@
...
+        "if (!(Test-Path \"$persist_dir\\vbam.ini.bak\")) {",
```
Comment (beyondmeat) [inline bucket/visualboyadvance-m.json:17]: test for the .ini not the .bak

File: bucket/visualboyadvance-m.json [hunk 4]
```diff
@@ -13,16 +13,33 @@
...
+        "   Set-Content \"$dir\\vbam.ini\" '[General]'",
+        "   Add-Content \"$dir\\vbam.ini\" 'OnlineUpdates=never'",
+        "   'Battery', 'Recording', 'Screenshot', 'State' | ForEach-Object {",
+        "   Add-Content -Path \"$dir\\vbam.ini\" -Value $_`Dir`=$_ }",
+        "} else {",
+        "   Copy-Item \"$persist_dir\\vbam.ini.bak\" \"$dir\\vbam.ini\"  -Force",
+        "}"
+    ],
+    "uninstaller": {
```
Comment (beyondmeat) [inline bucket/visualboyadvance-m.json:26]: can be dropped

File: bucket/visualboyadvance-m.json [hunk 5]
```diff
@@ -32,5 +49,6 @@
                 "url": "https://github.com/visualboyadvance-m/visualboyadvance-m/releases/download/v$version/visualboyadvance-m-Win-64bit.zip"
             }
         }
-    }
+    },
+    "notes": "Configuration files cannot be persisted, but will be retained during the update"
```
Comment (beyondmeat) [inline bucket/visualboyadvance-m.json:53]: can be dropped

**Comment by Calinou** [comment]
Superseded by https://github.com/Calinou/scoop-games/pull/596. Thanks for the contribution nonetheless!

**Pull request closed without merging**